scrapy/loader/__init__.py

```diff
@@ -24,16 +24,46 @@ class ItemLoader(object):
     default_output_processor = Identity()
     default_selector_class = Selector
 
-    def __init__(self, item=None, selector=None, response=None, **context):
+    def __init__(self, item=None, selector=None, response=None, parent=None, **context):
         if selector is None and response is not None:
             selector = self.default_selector_class(response)
         self.selector = selector
         context.update(selector=selector, response=response)
         if item is None:
             item = self.default_item_class()
-        self.item = context['item'] = item
         self.context = context
-        self._values = defaultdict(list)
+        self.parent = parent
+        self._local_item = context['item'] = item
+        self._local_values = defaultdict(list)
+
+    @property
+    def _values(self):
+        if self.parent is not None:
+            return self.parent._values
+        else:
+            return self._local_values
+
+    @property
+    def item(self):
+        if self.parent is not None:
+            return self.parent.item
+        else:
+            return self._local_item
+
+    def nested_loader(self, xpath=None, css=None):
+        if xpath is not None and css is not None:
+            raise ValueError("Cannot nest a loader with both a xpath selector and a css selector")
+
+        if xpath is not None:
+            selector = self.selector.xpath(xpath)
+
+        if css is not None:
+            selector = self.selector.css(css)
+
+        subloader = self.__class__(
+            item=self.item, selector=selector, parent=self
```

Comment:
I wasn't sure how to handle this since I haven't used contexts very much.  Would it be better for the nested loader to shared the parent's context or have a new, extended context?